### PR TITLE
Added login , register integration with Hermes server

### PR DESCRIPTION
**Pull request by lsimopoulos**

### CODE REVIEW

File: Web/hermes.client-web/src/App.vue [hunk 1]
```diff
@@ -1,16 +1,71 @@
...
+      <div v-if="!isLoggedIn" class="navbar-nav ml-auto">
+        <li class="nav-item">
+          <router-link to="/sign" class="nav-link">
+            <font-awesome-icon icon="user-plus" />
+            <span class="navbar-icon-text">Sign</span>
+          </router-link>
+        </li>
+      </div>
+
+      <div v-if="isLoggedIn" class="navbar-nav ml-auto">
```
Comment (pavanas) [inline Web/hermes.client-web/src/App.vue:38]:
```suggestion
      <div v-else class="navbar-nav ml-auto">
```

File: Web/hermes.client-web/src/services/auth.service.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      rejectUnauthorized: false,
+    });
+    return axios.post(API_URL + 'api/Users/Register/', {
+      username: email,
+      password: password
+    },{
+      httpsAgent : agent
+
+    })
+    .catch(function (error) {
```
Comment (pavanas) [inline Web/hermes.client-web/src/services/auth.service.js:54]:
```suggestion
    .catch((error) => {
```

File: Web/hermes.client-web/src/services/auth.service.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import axios from 'axios';
+
+const API_URL = 'https://localhost:7001/';
+
+class AuthService {
+  
+  login({email,password}) {
+    var body =
+    'username=' +
```
Comment (pavanas) [inline Web/hermes.client-web/src/services/auth.service.js:9]: Giati den ta stelneis san body object? 
```
body = {
    username: blah,
    password: blah,
    grant_type: blah,
    etc,
    etc..
}
```

File: Web/hermes.client-web/src/store/modules/auth.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ export const auth = {
+   namespaced: true,
+  state:  {
+    status: { loggedIn: false }
+  },
+  actions: {
+    attemptLogin({ commit }, {email,password}) {
+      return AuthService.login({email,password})
+      .then(
+        response => {
```
Comment (pavanas) [inline Web/hermes.client-web/src/store/modules/auth.js:12]: Mperdeytika edw. To `AuthService.login` kanei return eite to `response.data` eite `error`. Den kserw pws douleuei me 2 arguments :shrug:

File: Web/hermes.client-web/src/services/auth.service.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      headers: { 
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: 'Basic' + ' ' + btoa('chat_web_client:websuperdupersecret'),
+        Accept: 'application/json'
+      }})
+      .then(response => {
+        return response.data;
+      }) .catch((err) => {
+        console.log(err);
+    });
```
Comment (pavanas) [inline Web/hermes.client-web/src/services/auth.service.js:39]:
```suggestion
     const headers = {
        'Content-Type': 'application/x-www-form-urlencoded',
        Authorization: 'Basic' + ' ' + btoa('chat_web_client:websuperdupersecret'),
        Accept: 'application/json'
     };
     
     return axios.post(API_URL + 'connect/token', body, {headers});
```

File: Web/hermes.client-web/src/store/modules/auth.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          commit('loginSuccess', response);
+
+          return Promise.resolve(response);
+        },
+        error => {
+          commit('loginFailure');
+          return Promise.reject(error);
+        }
+      );
```
Comment (pavanas) [inline Web/hermes.client-web/src/store/modules/auth.js:21]: An thes to `attemptLogin` na gyrnaei kai ayto `promise` tote:
```suggestion
      return new Promise((resolve, reject) => {
         AuthService.login({email,password})
         .then(response => {
             commit('loginSuccess', response);
             resolve(response);
         }).catch((err) => {
             console.log(err);
             commit('loginFailure');
             reject(err);
         });
      })
```

File: Web/hermes.client-web/src/store/modules/auth.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    logout({ commit }) {
+      commit('logout');
+    },
+    attemptRegister({ commit }, {email,password}) {
```
Comment (pavanas) [inline Web/hermes.client-web/src/store/modules/auth.js:26]: To idio kai gia to `attemptRegister`

File: Web/hermes.client-web/src/services/auth.service.js (hunk 3 above)
Comment (lsimopoulos) [reply Web/hermes.client-web/src/services/auth.service.js:9]: https://identityserver4.readthedocs.io/en/latest/endpoints/token.html

**Pull request merged**